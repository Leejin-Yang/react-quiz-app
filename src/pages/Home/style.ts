import styled from 'styled-components';
import { responsive } from 'styles/constants';

export const Container = styled.main`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 0 20px;
`;

export const Section = styled.section`
  margin-bottom: 30px;
  line-height: 1.6;

  h2 {
    margin-bottom: 10px;
    font-size: 32px;
    font-weight: 600;
    font-style: italic;
  }
`;

export const ButtonWrapper = styled.div`
  display: flex;
  column-gap: 10px;

  @media ${responsive('MD')} {
    flex-direction: column;
    row-gap: 10px;
  }
`;
