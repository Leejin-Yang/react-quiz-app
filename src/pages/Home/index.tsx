import { useNavigate } from 'react-router-dom';

import Button from 'components/Button';

import { ButtonWrapper, Container, Section } from './style';

const Home = () => {
  const navigate = useNavigate();

  const onClick = () => {
    navigate('/quizzes');
  };

  return (
    <Container>
      <Section>
        <h2>Quiz</h2>
        <p>
          간단한 퀴즈를 풀어보세요❗️ <br /> 퀴즈는 영어로 제공됩니다 🙂
        </p>
      </Section>
      <ButtonWrapper>
        <Button onClick={onClick}>퀴즈 풀기</Button>
        <Button onClick={() => navigate('/study-note', { replace: true })}>오답 노트</Button>
      </ButtonWrapper>
    </Container>
  );
};

export default Home;
