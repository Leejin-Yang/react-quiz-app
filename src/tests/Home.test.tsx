import { fireEvent, render } from '@testing-library/react';

import Home from 'pages/Home';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUseNavigate,
}));

describe('<Home />', () => {
  it('matches snapshot', () => {
    const utils = render(<Home />);

    expect(utils.container).toMatchSnapshot();
  });
  it('has title and button', () => {
    const utils = render(<Home />);

    utils.getByText('Quiz');
    utils.getByText('퀴즈 풀기');
  });
  it('navigate to quiz page', () => {
    const utils = render(<Home />);
    const quizButton = utils.getByText('퀴즈 풀기');

    fireEvent.click(quizButton);
    expect(mockedUseNavigate).toHaveBeenCalledWith('/quizzes');
  });
});
