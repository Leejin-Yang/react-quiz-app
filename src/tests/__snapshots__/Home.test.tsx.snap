// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
<div>
  <main
    class="sc-gsnTZi etKRxi"
  >
    <section
      class="sc-dkzDqf gRJqEN"
    >
      <h2>
        Quiz
      </h2>
      <p>
        간단한 퀴즈를 풀어보세요❗️ 
        <br />
         퀴즈는 영어로 제공됩니다 🙂
      </p>
    </section>
    <div>
      <button
        class="sc-bczRLJ gDwxMx"
        type="button"
      >
        퀴즈 풀기
      </button>
    </div>
  </main>
</div>
`;
